pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1324,22 +1361,18 @@ func (npm *nominator) deleteNominatedPodIfExistsUnlocked(pod *v1.Pod) {
 // the pod. We update the structure before sending a request to update the pod
 // object to avoid races with the following scheduling cycles.
 func (npm *nominator) AddNominatedPod(logger klog.Logger, pi *framework.PodInfo, nominatingInfo *framework.NominatingInfo) {
-	npm.lock.Lock()
+	npm.nominatorLock.Lock()
 	npm.addNominatedPodUnlocked(logger, pi, nominatingInfo)
-	npm.lock.Unlock()
+	npm.nominatorLock.Unlock()
 }
 
 // NominatedPodsForNode returns a copy of pods that are nominated to run on the given node,
 // but they are waiting for other pods to be removed from the node.
 func (npm *nominator) NominatedPodsForNode(nodeName string) []*framework.PodInfo {
```

Comment:
Can we define `NominatedPodsForNode` fn in SchedulingQueue?
Then, I guess we don't need to pass `nominatedPodsToInfo` fn from SchedulingQueue to nominator.

Reply:
That was my initial idea, but it gives much overhead to change the nominator to scheduling queue, especially in tests, where we would need to inject the `nominatedPodsToInfo` anyway or redesign the tests deeper.

Reply:
The danger here is that, `nominatedPodsToInfo` takes qlock inside, while this `nominatedPodsToInfo` could be replacable within unit tests.
We could miss a deadlock scenario with unit tests running a mock `nominatedPodsToInfo` that doesn't take a qlock.

Reply:
Please make sure you create an issue for this after merging this PR so that we don't forget.